cli/cmdlineparser.cpp

```diff
@@ -1609,10 +1612,10 @@ void CmdLineParser::printHelp() const
     mLogger.printRaw(oss.str());
 }
 
-bool CmdLineParser::isCppcheckPremium() const {
-    if (mSettings.cppcheckCfgProductName.empty())
-        mSettings.loadCppcheckCfg();
-    return startsWith(mSettings.cppcheckCfgProductName, "Cppcheck Premium");
+bool CmdLineParser::isCppcheckPremium() {
+    Settings settings;
```

Comment:
@firewave May I ask you why do we need to create the local settings here?

Reply:
So we do not accidentally load the config multiple times times. That should be done in a deterministic way once.

Reply:
Okay agree on that one. But you only creating local variable so it's not working.
adding `static` to it will help though.

Reply:
> But you only creating local variable so it's not working. 

What is not working? We only need to compare s single string from the object. Also this is only done during the parsing of the command-line options.

> adding `static` to it will help though.

What? The method or the object?

Reply:
> What? The method or the object?

Both but as you've already marked the method you need to add it only here.

Reply:
> Both but as you've already marked the method you need to add it only here.

On the object it would not help as it still requires the function call.

I think I did look at the method but I am not sure. It is also just a limited usage.

Reply:
> What is not working?

In a blank Settings object the exename is not set and therefore it can't find the cppcheck.cfg file anymore after your changes.